src/xGadgetron/cGadgetron/include/sirf/Gadgetron/noncartesian_encoding.h

```diff
@@ -0,0 +1,142 @@
+A COPY OF THIS SOFTWARE AND ASSOCIATED DOCUMENTATION FILES (THE
+"SOFTWARE"), TO DEAL IN THE SOFTWARE WITHOUT RESTRICTION, INCLUDING
+WITHOUT LIMITATION THE RIGHTS TO USE, COPY, MODIFY, MERGE, PUBLISH,
+DISTRIBUTE, SUBLICENSE, AND/OR SELL COPIES OF THE SOFTWARE, AND TO
+PERMIT PERSONS TO WHOM THE SOFTWARE IS FURNISHED TO DO SO, SUBJECT TO
+THE FOLLOWING CONDITIONS:
+
+THE ABOVE COPYRIGHT NOTICE, THIS PERMISSION NOTICE, AND THE LIMITATION
+OF LIABILITY BELOW SHALL BE INCLUDED IN ALL COPIES OR REDISTRIBUTIONS
+OF SUBSTANTIAL PORTIONS OF THE SOFTWARE.
+
+SOFTWARE IS BEING DEVELOPED IN PART AT THE NATIONAL HEART, LUNG, AND BLOOD
+INSTITUTE, NATIONAL INSTITUTES OF HEALTH BY AN EMPLOYEE OF THE FEDERAL
+GOVERNMENT IN THE COURSE OF HIS OFFICIAL DUTIES. PURSUANT TO TITLE 17,
+SECTION 105 OF THE UNITED STATES CODE, THIS SOFTWARE IS NOT SUBJECT TO
+COPYRIGHT PROTECTION AND IS IN THE PUBLIC DOMAIN. EXCEPT AS CONTAINED IN
+THIS NOTICE, THE NAME OF THE AUTHORS, THE NATIONAL HEART, LUNG, AND BLOOD
+INSTITUTE (NHLBI), OR THE NATIONAL INSTITUTES OF HEALTH (NIH) MAY NOT
+BE USED TO ENDORSE OR PROMOTE PRODUCTS DERIVED FROM THIS SOFTWARE WITHOUT
+SPECIFIC PRIOR WRITTEN PERMISSION FROM THE NHLBI OR THE NIH.THE SOFTWARE IS
+PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR
+IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+*/
+
+/*!
+\file
+\ingroup SIRF non-cartesian encoding
+\brief Specification file for non-cartesian Fourier encoding.
+
+\author Johannes Mayer
+*/
+
+
+#ifndef NONCARTESIAN_ENCODING_H
+#define NONCARTESIAN_ENCODING_H
+
+#include <sirf/Gadgetron/fourierencoding.h>
+
+#include <gadgetron/hoNDArray.h>
+#include <gadgetron/vector_td.h>
+#include <gadgetron/vector_td_utilities.h>
+
+#include <gadgetron/hoNDFFT.h>
+#include <gadgetron/hoNFFT.h>
+
+
+namespace sirf{
+
+
+typedef Gadgetron::hoNDArray<Gadgetron::floatd2> GadgetronTrajectoryType2D;
+
+/*!
+\ingroup Gadgetron Extensions
+\brief Implementation to perform a non-cartesian FFT for RPE MR data
+*
+* backward(...): first data are sorted into a 4D matrix consisting of kx, ky, kz, coil
+* and then Fourier-transformed along the kx dimension. In the second step a NUFFT
+* along the two remaining dimensions is performed.
+*
+*/
+
+
+class RPEFourierEncoding : public FourierEncoding
+{
+public:
+    RPEFourierEncoding(): FourierEncoding() {}
+
+    virtual void forward(MRAcquisitionData& ac, const CFImage& img) const;
+    virtual void backward(CFImage& img, const MRAcquisitionData& ac) const;
+protected:
+    GadgetronTrajectoryType2D get_trajectory(const MRAcquisitionData& ac) const;
+
+};
+
+using namespace Gadgetron;
```

Comment:
please do not use `using` statement in `.h` files.  It means we could have name conflicts. (Feel free to use them in the `.cpp` of course).

Reply:
done